PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskSubJetFraction.cxx

```diff
@@ -1833,76 +1788,136 @@ void AliAnalysisTaskSubJetFraction::Terminate(Option_t *)
+  fJetRadius=fJetRadius*2.0;
   try {
-    fastjet::JetDefinition fJetDef(fastjet::antikt_algorithm, fJetRadius*2, static_cast<fastjet::RecombinationScheme>(0), fastjet::BestFJ30 );         
-    fastjet::ClusterSequence fClustSeqSA(fInputVectors, fJetDef);
-    std::vector<fastjet::PseudoJet> Modified_Jet=fClustSeqSA.inclusive_jets(0);
-    Modified_AliEmcalJet= new AliEmcalJet(Modified_Jet[0].perp(), Modified_Jet[0].pseudorapidity(), Modified_Jet[0].phi(), Modified_Jet[0].m());
+    fastjet::JetDefinition fJetDef(fastjet::antikt_algorithm, fJetRadius, static_cast<fastjet::RecombinationScheme>(0), fastjet::BestFJ30 );         
+    fClustSeqSA=new fastjet::ClusterSequence(fInputVectors, fJetDef);
+    Modified_Jet=fClustSeqSA->inclusive_jets(0);
   } catch (fastjet::Error) {
     AliError(" [w] FJ Exception caught.");
     //return -1;
   }
-  return Modified_AliEmcalJet;
-
-  
-				
+  fJetRadius=fJetRadius/2.0;
+  return Modified_Jet[0];				
 }
 
 
-std::vector<fastjet::PseudoJet> AliAnalysisTaskSubJetFraction::RandomiseTracks(std::vector<fastjet::PseudoJet> fInputVectors){
+std::vector<fastjet::PseudoJet> AliAnalysisTaskSubJetFraction::RandomiseTracks(AliEmcalJet *Jet,std::vector<fastjet::PseudoJet> fInputVectors){
+  fRandom= new TRandom3();
   fRandom->SetSeed(0);
   Double_t Random_Phi=0.0;
+  Double_t Random_Phi_Change=0.0;
   Double_t Random_Eta=0.0;
+  Double_t Random_Eta_Change=0.0;
   Double_t Track_Pt=0.0;
   std::vector<fastjet::PseudoJet> Random_Track_Vector;
   Random_Track_Vector.clear();
   for (Int_t i=0; i< fInputVectors.size(); i++){
-    Random_Phi=fRandom->Uniform(2*TMath::Pi());
-    Random_Eta=fRandom->Uniform(-0.9,0.9);
+    Random_Phi_Change=fRandom->Uniform(-1*fJetRadius,fJetRadius);
+    Random_Phi=Jet->Phi()+Random_Phi_Change;
+    Random_Eta_Change=fRandom->Uniform(-1*(TMath::Sqrt((fJetRadius*fJetRadius)-(Random_Phi_Change*Random_Phi_Change))),TMath::Sqrt((fJetRadius*fJetRadius)-(Random_Phi_Change*Random_Phi_Change)));
+    Random_Eta=Jet->Eta()+Random_Eta_Change;
     Track_Pt=fInputVectors[i].perp();
-    fastjet::PseudoJet Random_Track(Track_Pt*TMath::Cos(Random_Phi),Track_Pt*TMath::Sin(Random_Phi),Track_Pt*TMath::SinH(Random_Eta),(fInputVectors[i].m()*fInputVectors[i].m())+(Track_Pt*TMath::CosH(Random_Eta)*Track_Pt*TMath::CosH(Random_Eta)));
+    fastjet::PseudoJet Random_Track(Track_Pt*TMath::Cos(Random_Phi),Track_Pt*TMath::Sin(Random_Phi),Track_Pt*TMath::SinH(Random_Eta),TMath::Sqrt((Track_Pt*Track_Pt)+(Track_Pt*TMath::SinH(Random_Eta)*Track_Pt*TMath::SinH(Random_Eta))));
     Random_Track.set_user_index(i);
-    Random_Track_Vector.push_back(Random_Track); 
+    Random_Track_Vector.push_back(Random_Track);
   }
+  fInputVectors.clear();
   return Random_Track_Vector;
 }
 
 std::vector<fastjet::PseudoJet> AliAnalysisTaskSubJetFraction::AddExtraProng(std::vector<fastjet::PseudoJet> fInputVectors, Double_t Distance, Double_t PtFrac){
+  fRandom= new TRandom3();
```

Comment:
I guess the TRandom is used only locally, so it can be on the stack as well.

Reply:
Yes I think think in this case the stack should be ok. Do I need to create a new pull request to change this?

Reply:
You can attach it to the current pull request. All patches you commit in your local branch and push to your fork will be included in the current pull request.